source/plugin/messagechannels.rst

```diff
@@ -0,0 +1,118 @@
+transforming. The channel will then transform the message for every recipient and send the transformed message.
+
+Extended Application: Chat Channels
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Message channels have a very useful application as they can be used to establish chat channels. For example, you could
+establish a message channel for every chat channel you wish to have. Then, when a ``MessageReceiver`` joins a channel,
+such as with ``/join <channel name>``, simply set the ``MessageReceiver``'s ``MessageChannel`` to the appropriate c
+hannel using the ``MessageReceiver#setMessageChannel(MessageChannel channel)``. This will cause every message sent
+*from* the ``MessageReceiver`` to be passed to the given ``MessageChannel`` instead of the default one. Alternatively,
+you could listen to ``MessageChannelEvent``, and set the appropriate ``MessageChannel`` using
+``MessageChannelEvent#setChannel(MessageChannel channel)``. Passing ``null`` to that method will unset any custom
+channel, causing the message to be sent to the originalMessageChannel instead.
+
+Transforming Messages
+=====================
+
+You can apply a filter to all ``Text``\ s that pass through a ``MessageSink`` to change the message however you like.
+This is possible by extending ``MessageChannel`` and overriding the ``transformMessage()`` method as demonstrated
+below.
+
+**Example: Transforming Messages**
+
+The following code excerpt defines a ``AdminMessageSink`` class which overrides the default ``transformMessage`` method.
+The new transformation method will first check if a source exists and, if it exists and is a permission ``Subject``,
+has the ``myplugin.admin`` permission. If that is the case, the text will be prefixed with a red ``[Admin]``.
+
+.. code-block:: java
+
+    // TODO Update code example
+
+
+Note that we do not wish to define any additional recipients, so we return an empty collection in the ``getMembers``
+method.
+
+Thanks to the capabilities of combined ``MessageChannel``\ s, we can combine our newly created ``AdminMessageChannel``
+with any other ``MessageChannel``. Now if a player joins that has the ```myplugin.admin`` permission, we will obtain
+the ``MessageChannel`` his messages are sent to, combine it with an ``AdminMessageChannel`` and set the combined
+channel back on the player. That way all his messages are sent to everyone specified in the original channel, but
+due to the addition of the ``AdminMessageChannel`` will be prefixed.
+
+.. code-block:: java
+
+    import org.spongepowered.api.event.Listener;
+    import org.spongepowered.api.event.network.ClientConnectionEvent;
+
+    @Listener
+    public void onClientConnectionJoin(ClientConnectionEvent.Join event) {
+        Player player = event.getEntity();
+
+        MessageChannel originalChannel = player.getChannel();
+        MessageChannel newChannel = MessageChannel.combined(originalChannel, new AdminMessageChannel());
```

Comment:
Instead of creating a new instance of `AdminMessageChannel` you can store it in the example class